jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -16,203 +16,214 @@
-        if(!isNullOrEmpty(userName) && !isNullOrEmpty(groupName)){
-            caseService.call(
-                    (Void) -> findCaseInstance()
-            ).assignGroupAndUserToRole(serverTemplateId, containerId, caseId, roleName, userName,groupName);
-        }else {
-            if (!isNullOrEmpty(userName)) {
-                addUserToRole(userName, roleName);
-            }
-            if(!isNullOrEmpty(groupName)){
-                addGroupToRole(groupName,roleName);
-            }
+    public void editAction(CaseRoleAssignmentSummary caseRoleAssignmentSummary) {
+
+        editRoleAssignmentView.setValue(CaseRoleAssignmentSummary.builder()
+                .name(caseRoleAssignmentSummary.getName())
+                .users(caseRoleAssignmentSummary.getUsers())
+                .groups(caseRoleAssignmentSummary.getGroups()).build());
+        editRoleAssignmentView.show(() -> assignToRole());
+    }
+
+    protected CaseRoleAssignmentSummary getRoleAssignment(final CaseInstanceSummary cis, final String roleName) {
+        if (CASE_OWNER_ROLE.equals(roleName)) {
+            return CaseRoleAssignmentSummary.builder().name(roleName).users(Arrays.asList(cis.getOwner())).build();
         }
+        return cis.getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(CaseRoleAssignmentSummary.builder().name(roleName).build());
     }
 
-    protected void addUserToRole(final String userName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignUserToRole(serverTemplateId, containerId, caseId, roleName, userName);
+    protected void assignToRole() {
+        final List<String> assignmentErrors =
+                caseRolesValidations.validateRolesAssignments(caseDefinition,
+                        Arrays.asList(editRoleAssignmentView.getValue()));
+        if (assignmentErrors.isEmpty() == false) {
+            editRoleAssignmentView.showValidationError(assignmentErrors);
+            return;
+        }
+        editRoleAssignmentView.hide();
+        storeRoleAssignments(editRoleAssignmentView.getValue().getName(),
+                editRoleAssignmentView.getValue().getUsers(),
+                editRoleAssignmentView.getValue().getGroups());
     }
 
-    protected void addGroupToRole(final String groupName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignGroupToRole(serverTemplateId, containerId, caseId, roleName, groupName);
+    protected CaseRoleAssignmentSummary getRoleAssignmentByName(String roleName) {
+        return allUnfilteredElements.stream()
+                .filter(caseRoleAssignmentSummary -> roleName.equals(caseRoleAssignmentSummary.getName()))
+                .findFirst().get();
+    }
+
+    protected void storeRoleAssignments(String roleName, List<String> users, List<String> groups) {
```

Comment:
Sorry,  I understand now pass the model instead the roleName.. yes, I'll do that change